components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/IdentitySCIMManager.java

```diff
@@ -88,13 +89,19 @@ private void init() throws CharonException {
         // This is necessary to instantiate here as we need to encode exceptions if they occur.
         encoder = new JSONEncoder();
 
-        // Define endpoint urls to be used in Location Header.
-        endpointURLs.put(SCIMConstants.USER_ENDPOINT, SCIMCommonUtils.getSCIMUserURL());
-        endpointURLs.put(SCIMConstants.GROUP_ENDPOINT, SCIMCommonUtils.getSCIMGroupURL());
-        endpointURLs.put(SCIMConstants.SERVICE_PROVIDER_CONFIG_ENDPOINT, SCIMCommonUtils
-                .getSCIMServiceProviderConfigURL());
-        endpointURLs.put(SCIMConstants.RESOURCE_TYPE_ENDPOINT, SCIMCommonUtils.getSCIMResourceTypeURL());
-
+        if (!IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+            // Define endpoint urls to be used in Location Header.
+            endpointURLs.put(SCIMConstants.USER_ENDPOINT, SCIMCommonUtils.getSCIMUserURL());
+            endpointURLs.put(SCIMConstants.GROUP_ENDPOINT, SCIMCommonUtils.getSCIMGroupURL());
+            endpointURLs.put(SCIMConstants.SERVICE_PROVIDER_CONFIG_ENDPOINT, SCIMCommonUtils
+                    .getSCIMServiceProviderConfigURL());
+            endpointURLs.put(SCIMConstants.RESOURCE_TYPE_ENDPOINT, SCIMCommonUtils.getSCIMResourceTypeURL());
+        } else {
```

Comment:
If we do the suggested change below we can get rid of this part as well. Just the ServiceURLBuilder is enough for both modes.